### ISSUE TITLE
Reload option doesnt work with unix sockets

### ISSUE THREAD

**Issue by euri10** [Kludex/uvicorn#722]
Was about to report the same thing, but with unix sockets.

_Originally posted by @magnusjjj in https://github.com/encode/uvicorn/issues/553#issuecomment-663530183_

**Comment by magnusjjj** [comment]
Yeah, uh, I have looked through the source.

main.py:346 
```
    if config.should_reload:
        sock = config.bind_socket()
        supervisor = ChangeReload(config, target=server.run, sockets=[sock])
        supervisor.run()
    elif config.workers > 1:
        sock = config.bind_socket()
        supervisor = Multiprocess(config, target=server.run, sockets=[sock])
        supervisor.run()
    else:
        server.run()
```

That. It seems like there are three kinds of 'processes', and they can't "stack".

The problem lies in that unix sockets, and passing a slew of premade sockets, require different calls to setup loops with asyncio, if I read the code correctly, as seen in main.py:445

So it's the same bug for both having workers and the reloader.
Creation of the sockets should *probably* be moved to the config.bind_socket (or rather, bind_sockets i guess), and then later code figuring out whether or not the sockets given to them are unix sockets or workers. This is my first glance at the code though.

**Comment by euri10** [comment]
yes you're right I think this would need a refactor, a good example would be in hypercorn I think.
lacking time to tackle on this though

**Comment by Kludex** [comment]
It looks like this was fixed by #1009.
